sys-process/bottom: RUSTSEC-2024-0019: Add 0.10.1

bowlofeggs opened the pull request:
This change adds bottom 0.10.1.

This change addresses RUSTSEC-2024-0019[0][1]. The advisory states that only Windows is affected.

Closes: https://bugs.gentoo.org/937026

[0] https://rustsec.org/advisories/RUSTSEC-2024-0019
[1] https://github.com/ClementTsang/bottom/issues/1521

<!-- Please put the pull request description above -->

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

mgorny commented on sys-process/bottom/bottom-0.9.7.ebuild:
```diff
@@ -0,0 +1,243 @@
+HOMEPAGE="https://github.com/ClementTsang/bottom"
+SRC_URI="${CARGO_CRATE_URIS}"
+
+LICENSE="0BSD Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD-2 Boost-1.0 ISC MIT MPL-2.0 Unicode-DFS-2016 Unlicense ZLIB"
```
Any reason it's not using split license per `pycargoebuild`?

bowlofeggs replied:
I used `cargo ebuild` to generate this (and really, I just used it to generate the big list of crates this round, and copied the code from version 0.9.6 otherwise…)

When you say split license, do you mean like one per line? Or is it something else?

bowlofeggs replied:
I ran pycargoebuild, and I think I see what you mean. Like this?

```
LICENSE="MIT"
# Dependent crate licenses
LICENSE+=" Apache-2.0 ISC MIT MPL-2.0 Unicode-DFS-2016"
```

I do wonder why it comes up with a different list than `cargo ebuild` came up with though. I think I might keep the list from `cargo ebuild` but use this style if that's OK with you?

bowlofeggs replied:
In https://github.com/gentoo/gentoo/compare/4926fde527c8a2fc13c8d9983918691107cbcf1c..00129d6a07b1c795b455f951f6bc6827abefef0d I changed it to use the style I see from `pycargoebuild`, but the list I see from `cargo ebuild`.

juippis replied:
Also in general **cargo ebuild** is unmaintained and will probably stop working at some point. Better switch to a maintained alternative sooner than later.

bowlofeggs replied:
I pushed up https://github.com/gentoo/gentoo/pull/37889/commits/51c5415ac073f7bb46eca8adfa4abd0c24ae891d which has the license set from `pycargoebuild`.

I'll also adjust my gentoo dev environment to replace cargo ebuild.

mgorny commented on sys-process/bottom/bottom-0.9.7.ebuild:
```diff
@@ -0,0 +1,243 @@
+	# Stripping symbols should be the choice of the user.
+	sed -i '/strip = "symbols"/d' Cargo.toml || die "Unable to patch out symbol stripping"
+
+	eapply_user
```
That's part of `default`.

bowlofeggs replied:
Dropped in https://github.com/gentoo/gentoo/compare/e86e64453754b4f03ab5840950e5db5cd66aa210..00073e8f2704e99b04b90ab283a525ad3d04383f

mgorny commented on sys-process/bottom/bottom-0.9.7.ebuild:
```diff
@@ -0,0 +1,243 @@
+}
+
+src_configure() {
+	myfeatures=(
```
`local`

bowlofeggs replied:
Fixed in https://github.com/gentoo/gentoo/compare/00073e8f2704e99b04b90ab283a525ad3d04383f..1e795f19173d79b8ce9623162e752e194205150e

mgorny commented on sys-process/bottom/bottom-0.9.7.ebuild:
```diff
@@ -0,0 +1,243 @@
+
+	# Find generated shell completion files. btm.bash can be present in multiple dirs if we build
+	# additional features, so grab the first match only.
+	local BUILD_DIR="$(dirname $(find target -name btm.bash -print -quit || die) || die)"
```
This doesn't sound like the global `BUILD_DIR` var, so make it lowercase to avoid accidental collision with something.

bowlofeggs replied:
Fixed in https://github.com/gentoo/gentoo/compare/1e795f19173d79b8ce9623162e752e194205150e..4926fde527c8a2fc13c8d9983918691107cbcf1c

bowlofeggs commented:
Since I opened this PR, there has been two more bottom releases. I personally think it's OK to add those in a later PR, but if you prefer I can change this PR to target the latest release instead.

It's ready for another look. Thanks for the reviews!

mgorny commented:
Please always update to the newest (unless there's a good reason not to). I hope the new version removes the need for patching. If not, is there a bug reported upstream?

bowlofeggs commented:
> Please always update to the newest (unless there's a good reason not to). I hope the new version removes the need for patching. If not, is there a bug reported upstream?

I've rebased onto 0.10.1 now.

I had previously filed https://github.com/ClementTsang/bottom/issues/1521 and https://github.com/ClementTsang/bottom/issues/1500, but they are both fixed in 0.10.1 so the patches are now dropped.

mgorny commented:
Thanks. I'm going to test it now, and push if it works.
